edenai_apis/apis/leonardo/leonardo_api.py

```diff
@@ -0,0 +1,142 @@
+
+    @overload
+    def __get_response(
+        self, url: str, payload: dict
+    ) -> Generator: ...
+
+    @overload
+    def __get_response(
+        self, url: str, payload: dict
+    ) -> dict: ...
+
+    def __get_response(
+        self, url: str, payload: dict
+    ) -> Union[Generator, dict]:
+        # Launch job
+        launch_job_response = requests.post(url, headers=self.headers, json=payload)
+        
+        try:
+            launch_job_response_dict = launch_job_response.json()
+        except requests.JSONDecodeError:
+            raise ProviderException(
+                launch_job_response.text, code=launch_job_response.status_code
+            )
+        if launch_job_response.status_code != 200:
+            raise ProviderException(
+                launch_job_response_dict.get("error", launch_job_response_dict),
+                code=launch_job_response.status_code
+            )
+
+        generation_id = launch_job_response_dict["sdGenerationJob"]["generationId"]
+        url_get_response = f"{self.base_url}/generations/{generation_id}"
+
+        # Get job response
+        response = requests.get(url_get_response, headers=self.headers)
+
+        if response.status_code >= 500:
+            raise ProviderException(
+                message=http.client.responses[response.status_code],
+                code=response.status_code,
+            )
+        try:
+            response_dict = response.json()
+        except requests.JSONDecodeError:
+            raise ProviderException(f"Invalid JSON response: {response.text}")
+        
+        if response.status_code != 200:
+            raise ProviderException(
+                response_dict.get("detail"), code=response.status_code
+            )
+        
+        status = response_dict["generations_by_pk"]["status"]
+        while status != "COMPLETE":
+            response = requests.get(url_get_response, headers=self.headers)
+            try:
+                response_dict = response.json()
+            except requests.JSONDecodeError:
+                raise ProviderException(response.text, code=response.status_code)
+
+            if response.status_code != 200:
+                raise ProviderException(
+                    response_dict.get("error", response_dict), code=response.status_code
+                )
+
+            status = response_dict["generations_by_pk"]["status"]
+
+        return response_dict
+
+    def image__generation(
+        self,
+        text: str,
+        resolution: Literal["256x256", "512x512", "1024x1024"],
+        num_images: int = 1,
+        model: Optional[str] = None,
+    ) -> ResponseType[GenerationDataClass]:
+        size = resolution.split("x")
+        payload = {
+            "prompt": text,
+            "width": int(size[0]),
+            "height": int(size[1]),
+            "modelId": get_model_id_image.get(model, model),
+            "num_images": num_images,

```

Comment:
I'm not sure if this comment is relevant, as we should already limit in the back the number of images that a user can retrieve, but I think it would be good if we limit here also in case of